assets/pogo/preferences/cmake_tango.opt.in

```diff
@@ -0,0 +1,65 @@
+#
+#	Set default PATH
+#
+set(EXECUTABLE_OUTPUT_PATH "${CMAKE_BINARY_DIR}/bin/")
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${EXECUTABLE_OUTPUT_PATH})
+
+if(DEFINED TANGO_HOME)
+	message("-DTANGO_HOME specified in the command line")
+else()
+	set(TANGO_HOME $ENV{TANGO_HOME})
+endif(DEFINED TANGO_HOME)
+
+if(DEFINED TANGO_HOME)
+	message("TANGO_HOME = ${TANGO_HOME}")
+else()
+	set(TANGO_HOME @prefix@)
+endif(DEFINED TANGO_HOME)
+
+set(ENV{PKG_CONFIG_PATH} "${TANGO_HOME}/lib/pkgconfig:$ENV{PKG_CONFIG_PATH}")
+
+if(DEFINED OMNI_HOME)
+	message("-DOMNI_HOME specified in the command line")
+else()
+	set(OMNI_HOME $ENV{OMNI_HOME})
+endif(DEFINED OMNI_HOME)
+if(DEFINED OMNI_HOME)
+	message("OMNI_HOME = ${OMNI_HOME}")
+	set(ENV{PKG_CONFIG_PATH} "${OMNI_HOME}/lib/pkgconfig:$ENV{PKG_CONFIG_PATH}")
+endif(DEFINED OMNI_HOME)
+
+
+if(DEFINED ZMQ_HOME)
+	message("-DZMQ_HOME specified in the command line")
+else()
+	set(ZMQ_HOME $ENV{ZMQ_HOME})
+endif(DEFINED ZMQ_HOME)
+if(DEFINED ZMQ_HOME)
+	message("ZMQ_HOME = ${ZMQ_HOME}")
+	set(ENV{PKG_CONFIG_PATH} "${ZMQ_HOME}/lib/pkgconfig:$ENV{PKG_CONFIG_PATH}")
+endif(DEFINED ZMQ_HOME)
+
+#
+# Compiler and linker flags
+#
+include(CheckCXXCompilerFlag)
+CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
+CHECK_CXX_COMPILER_FLAG("-std=c++0x" COMPILER_SUPPORTS_CXX0X)
+if(COMPILER_SUPPORTS_CXX11)
+	set(C++11_FLAGS "-std=c++11")
+elseif(COMPILER_SUPPORTS_CXX0X)
+	set(C++11_FLAGS "-std=c++0x")
+else()
+        message(STATUS "The compiler ${CMAKE_CXX_COMPILER} has no C++11 support.")
+endif()
+
+set(CMAKE_CXX_FLAGS "${CXXFLAGS_USER}  ${CMAKE_CXX_FLAGS} ${C++11_FLAGS}")
```

Comment:
Consider removing `C++11_FLAGS`. This is harmful as all compilers are using C++14 by default from some time.

Following workaround does not work as `CMAKE_CXX_FLAGS` goes before `C++11_FLAGS` and only last `-std=` flag is effective.
```
cmake .. -DCMAKE_CXX_FLAGS='-std=c++17'
```

Reply:
I see. Thanks for the input.
So you propose to use directly the default provided by the compiler and if the user wants to add -std option, this can be added via the CXXFLAGS_USER (in the device server Makefile) or via CMAKE_CXX_FLAGS CMake variable?

Reply:
Yes, I suggest to remove any `-std=` options. It seems to be unnecessary limitation (for gcc 6 or better), as there were efforts in the past to compile tango with c++17 (tango-controls/cppTango#405).

Tango does not require c++11 so the flag is not helping also in gcc 5 case (which defaults to c++98).

Reply:
I removed -std flag from cmake_tango.opt and from tango.opt (for consistency) in the following commits:
- https://github.com/tango-controls/TangoSourceDistribution/pull/36/commits/cd6e45bdb9027c3de527953ab2ce083f5834dec7
- https://github.com/tango-controls/TangoSourceDistribution/pull/36/commits/33b91987c51087d5eb201a44b208ea327eb16121

Reply:
Thanks! I've just checked the new `tango.opt`. A device server (benchmark) compiles just fine without `-std` on gcc 7. It also compiles with `-std=c++17`.

Reply:
I've checked with gcc 4.4 (on an old Debian 6) and it requires the flag -std=c++0x or you would get this error message:
```
In file included from /usr/include/c++/4.4/type_traits:35,
                 from /tmp/Reynald/install/debian6/include/tango/devapi_attr.tpp:38,
                 from /tmp/Reynald/install/debian6/include/tango/tango.h:102,
                 from ./Bidon.h:36,
                 from Bidon.cpp:37:
/usr/include/c++/4.4/c++0x_warning.h:31:2: error: #error This file requires compiler and library support for the upcoming ISO C++ standard, C++0x. This support is currently experimental, and must be enabled with the -std=c++0x or -std=gnu++0x compiler options.
```

I don't have the problem when compiling with CMake  on Debian 6 (of course you need to install a recent CMake version) because it uses pkgconfig. 
tango.pc file has this line when installed on Debian6:
`Cflags: -std=c++0x -I${includedir}/tango`

On Linux Mint 19 (with gcc 7.3) it is:
`Cflags:  -I${includedir}/tango`

I think we could put something in place similar to what was done in this PR for cmake_tango.opt where we use the flags given by tango.pc file.
The current tango.opt file is already using pkg-config to get the tango LD_FLAGS, but not for the CXXFlags.
Moreover, it seems like it is not needed to use pkg-config to get the LDFLAGS needed for OmniORB and zmq since `pkg-config --libs tango` seems to already provide that information.

I personally think that it would also be useful to set default output dir to bin instead of $(HOME)/DeviceServers for device servers compilations in tango.opt because I find it very confusing for newcomers (and experimented users too actually).

Reply:
Where did you get those .pc files?
* There is one in cppTango which does not specify any flags except `-I`:
  https://github.com/tango-controls/cppTango/blob/tango-9-lts/tango.pc.cmake
* And there is one in TangoSourceDistribution which uses either `c++11`, `c++0x` or none:
  https://github.com/tango-controls/TangoSourceDistribution/blob/master/assets/lib/cpp/tango.pc.in

The flag in source distribution is set according to the compiler version used to compile Tango itself. Device server may be compiled later with some other version.

Maybe it will be enough to add `-std=c++0x` in tango.opt if GCC version is <5 (6?) or >4.3?

> I personally think that it would also be useful to set default output dir to bin instead of $(HOME)/DeviceServers for device servers compilations in tango.opt because I find it very confusing for newcomers (and experimented users too actually).

I think it's a good idea. Maybe some `install` target which will put the server in `~/.local/bin` or `/usr/local/bin` will be also useful.

Reply:
> Where did you get those .pc files?

I got the one from TangoSourceDistribution

> The flag in source distribution is set according to the compiler version used to compile Tango itself. Device server may be compiled later with some other version.

But then the user would need to update PKG_CONFIG_PATH to take into account the tango.pc file from this other version, and the good flags will be used...
> 
> Maybe it will be enough to add `-std=c++0x` in tango.opt if GCC version is <5 (6?) or >4.3?
> 
> > I personally think that it would also be useful to set default output dir to bin instead of $(HOME)/DeviceServers for device servers compilations in tango.opt because I find it very confusing for newcomers (and experimented users too actually).
> 
> I think it's a good idea. Maybe some `install` target which will put the server in `~/.local/bin` or `/usr/local/bin` will be also useful.

Actually, there is already an install target (at least in the makefile version):
https://github.com/bourtemb/TangoSourceDistribution/blob/fix-34/assets/pogo/preferences/common_target.opt#L171

It installs the exe in $TANGO_HOME/bin